File: 3453-generate-binary-strings-without-adjacent-zeros/3453-generate-binary-strings-without-adjacent-zeros.py
class Solution:
    def validStrings(self, n: int) -> List[str]:
        lis = []

        def f(s: str):
            nonlocal lis
            if len(s) >= n:
                lis.append(s)
                return
            if s and s[-1] == '0':
                f(s + '1')
            else:
                f(s + '0')
                f(s + '1')
        
        f("")
        return lis

File: 3453-generate-binary-strings-without-adjacent-zeros/3453-generate-binary-strings-without-adjacent-zeros.cpp
class Solution {
public:
    vector<string> validStrings(int n) {
        vector<string> ans;
        function<void (int, int, string)> f = [&] (int i, int j, string temp) -> void {
            if(i == n) {
                ans.push_back(temp);
                return;
            }
            if(j == 0) {
                f(i + 1, j + 1, temp + "0");
            }
            f(i + 1, 0, temp + "1");
        };
        f(0, 0, "");

        return ans;
    }
};